dotnet/src/SemanticKernel.Skills/Skills.Memory.Sqlite/Database.cs

```diff
@@ -43,6 +44,42 @@ public static async Task<SqliteConnection> CreateConnectionAsync(string filename
         await cmd.ExecuteNonQueryAsync(cancel);
     }
 
+    public static async Task InsertOrIgnoreAsync(this SqliteConnection conn,
+    string collection, string key, string? value, string? timestamp, CancellationToken cancel = default)
+    {
+        await CreateTableAsync(conn, cancel);
+
+        //Could potentially use REPLACE instead of IGNORE here
+        SqliteCommand cmd = conn.CreateCommand();
+        cmd.CommandText = $@"
+             INSERT or IGNORE into {TableName}(collection, key, value, timestamp)
+             VALUES(@collection, @key, @value, @timestamp); ";
+
+        cmd.Parameters.AddWithValue("@collection", collection);
+        cmd.Parameters.AddWithValue("@key", key);
+        cmd.Parameters.AddWithValue("@value", value ?? string.Empty);
+        cmd.Parameters.AddWithValue("@timestamp", timestamp ?? string.Empty);
+        await cmd.ExecuteNonQueryAsync(cancel);
+    }
+
+    public static async Task InsertOrReplaceAsync(this SqliteConnection conn,
```

Comment:
I believe InsertOrReplace is commonly known as "Upsert" in the database biz. Let's stick with that unless there is a difference here.

Reply:
Technically both INSERT or IGNORE and INSERT and REPLACE are UPSERTs; I provided options for both. So i could rename wasnt sure if it made sense just to rename one of them.

Reply:
"INSERT or IGNORE" is different from "UPSERT", no?